### PR TITLE
replace manual hwrng seeding with hwrng entropy validation

### PR DESCRIPTION
**Pull request by lrvick**
- [x] finish implementation
- [x] test on live enclave

### CODE REVIEW

File: src/qos_system/src/lib.rs [hunk 1]
```diff
@@ -135,39 +144,36 @@ pub fn socket_connect(
...
+/// Verify expected hwrng is loaded
+pub fn check_hwrng(rng_expected: &str) -> Result<(), SystemError> {
+	let filename: &str = "/sys/class/misc/hw_random/rng_current";
+	let mut fd = match File::open(filename) {
+		Ok(fd) => fd,
+		Err(e) => {
+			return Err(SystemError {
+				message: format!("Failed to open {}", &filename),
+			})
+		}
```
Comment (emostov) [inline src/qos_system/src/lib.rs]:
```suggestion
	let mut fd = File::open(filename)
	    .map_err(SystemError { message: format!("Failed to open {}", &filename))?;
```

File: src/qos_system/src/lib.rs [hunk 2]
```diff
@@ -135,39 +144,36 @@ pub fn socket_connect(
...
-	// We just simply write samples to /dev/urandom now.
-	// See: https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.10.119
-	match random_fd.write_all(&entropy_sample) {
-		Ok(()) => Ok(entropy_sample.len()),
-		Err(_) => {
+	let mut rng_current = String::new();
+	match fd.read_to_string(&mut rng_current) {
+		Ok(s) => s,
+		Err(e) => {
 			return Err(SystemError {
```
Comment (emostov) [inline src/qos_system/src/lib.rs]:
```suggestion
	fd.read_to_string(&mut rng_current)
	    .map_err(SystemError { message: format!("Failed to read {}", &filename) })?
```

File: src/qos_system/src/lib.rs [hunk 3]
```diff
@@ -135,39 +144,36 @@ pub fn socket_connect(
...
-				message: String::from("Failed to write to /dev/urandom"),
-			});
+				message: format!("Failed to read {}", &filename),
+			})
```
Comment (emostov) [inline src/qos_system/src/lib.rs]:
```suggestion
```

File: src/qos_system/src/lib.rs [hunk 4]
```diff
@@ -135,39 +144,36 @@ pub fn socket_connect(
...
-	use std::io::Write;
-	use std::fs::OpenOptions;
-
-	let entropy_sample = match source(size) {
-		Ok(sample) => sample,
-		Err(e) => return Err(e),
+/// Verify expected hwrng is loaded
+pub fn check_hwrng(rng_expected: &str) -> Result<(), SystemError> {
+	let filename: &str = "/sys/class/misc/hw_random/rng_current";
+	let mut fd = match File::open(filename) {
```
Comment (emostov) [inline src/qos_system/src/lib.rs]: For opening the file and reading to string you could just do it in one line with

```
let rng_current = std::fs::read_to_string(filename)
    .map_err(SystemError { message: format!("Failed to open {}", &filename) })?;
```

File: src/qos_system/src/lib.rs (hunk 3 above)
Comment (lrvick) [reply src/qos_system/src/lib.rs]: This looks like no change at all. Even if it is whitespace the before and after look identical in this view.

File: src/qos_system/src/lib.rs (hunk 4 above)
Comment (emostov) [reply src/qos_system/src/lib.rs]: that would get rid of most code below

File: src/qos_system/src/lib.rs (hunk 3 above)
Comment (emostov) [reply src/qos_system/src/lib.rs]: i just meant it will be covered in the `map_err` now

**Pull request merged**